File: 14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& st) {
        string prefix = st[0];

        for (int i = 1; i < st.size(); i++) {
            string currentString = st[i];

            while (currentString.find(prefix) != 0) {
                prefix = prefix.substr(0, prefix.size() - 1);
                if (prefix.size() == 0)
                    return "";
            }
        }

        return prefix;
    }
};

File: 14. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] st) {

        String prefix = st[0];

        for (int i = 1; i < st.length; i++) {
            String currentString = st[i];

            while (currentString.indexOf(prefix) != 0) {
                prefix = prefix.substring(0, prefix.length() - 1);
                if (prefix.length() == 0) {
                    return "";
                }
            }
        }

        return prefix;
    }
}
